### ISSUE TITLE
CRAN check ERROR and risk of archiving

### ISSUE THREAD

**Issue by psolymos** [atahk/pscl#7]
Hi @simonjackman and @atahk . As a maintainer of one of pscl's strong revdep packages (opticut) I have received an email from CRAN that pscl is on its way to being archived if [errors](https://www.r-project.org/nosvn/R.check/r-devel-linux-x86_64-debian-clang/pscl-00check.html) are not fixed:

```
Error in if (class(object) == "rollcall") mat <- object$votes :
  the condition has length > 1
Calls: rollcall -> checkVotes
Execution halted
```

I believe you have fixed the issue already (`checkVotes` checking object class that is a vector), because  now I see that minor version is ahead of CRAN version (from 2017) and the code now looks like `if(class(object)[1]=="rollcall")`. Although this option certainly resolves the error, but it might have undesired consequences if the object class is e.g. `c("something", "rollcall", "something_else")` in which case it still inherits from the rollcall class. So a better solution would be `if(inherits(object, "rollcall"))`.

I just wanted to open this ticket and inquire about the status of possible (soon-ish) CRAN submission, and if there is any way I can be of help to facilitate that happening.

pscl is a really awesome package, so let's keep it on CRAN!

**Comment by simonjackman** [comment]
Pushed an update up to CRAN.  Should be live shortly.

Thanks!

Professor Simon Jackman
Chief Executive Officer,
United States Studies Centre,
University of Sydney,
Sydney, NSW
Australia

Sent from an iPhone (so please excuse any terseness)

On 20 Feb 2020, at 3:28 am, Peter Solymos <notifications@github.com> wrote:

﻿

Hi @simonjackman<https://github.com/simonjackman> and @atahk<https://github.com/atahk> . As a maintainer of one of pscl's strong revdep packages (opticut) I have received an email from CRAN that pscl is on its way to being archived if errors<https://www.r-project.org/nosvn/R.check/r-devel-linux-x86_64-debian-clang/pscl-00check.html> are not fixed:

Error in if (class(object) == "rollcall") mat <- object$votes :
  the condition has length > 1
Calls: rollcall -> checkVotes
Execution halted


I believe you have fixed the issue already (checkVotes checking object class that is a vector), because now I see that minor version is ahead of CRAN version (from 2017) and the code now looks like if(class(object)[1]=="rollcall"). Although this option certainly resolves the error, but it might have undesired consequences if the object class is e.g. c("something", "rollcall", "something_else") in which case it still inherits from the rollcall class. So a better solution would be if(inherits(object, "rollcall")).

I just wanted to open this ticket and inquire about the status of possible (soon-ish) CRAN submission, and if there is any way I can be of help to facilitate that happening.

**Comment by psolymos** [comment]
Thanks, @simonjackman !

**Comment by atahk** [comment]
Version 1.5.5 just made it to CRAN today and should work properly with R 4.0. Pull request #9 is now merged (though not before 1.5.5) and should future-proof by moving everything to `inherits`. Thanks, @psolymos
